sdk/storage/azure-storage-queues/README.md

```diff
@@ -0,0 +1,158 @@
+# Azure Storage Queues Client Library for C++
+
+Azure Queue storage is a service for storing large numbers of messages that can be accessed from anywhere in the world via authenticated calls using HTTP or HTTPS. A single queue message can be up to 64 KB in size, and a queue can contain millions of messages, up to the total capacity limit of a storage account.
+
+## Getting started
+
+### Install the package
+
+The easiest way to acquire the C++ SDK is leveraging vcpkg package manager. See the corresponding [Azure SDK for C++ readme section][azsdk_vcpkg_install].
+
+To install Azure Storage packages via vcpkg:
+
+```batch
+vcpkg install azure-storage-queues-cpp
+```
+
+Then, use in your CMake file:
+
+```CMake
+find_package(azure-storage-queues-cpp CONFIG REQUIRED)
+target_link_libraries(<your project name> PRIVATE Azure::azure-storage-queues)
+```
+
+### Prerequisites
+
+You need an Azure subscription and a [Storage Account][storage_account_overview] to use this package.
+
+To create a new Storage Account, you can use the [Azure Portal][create_account_with_azure_portal], [Azure PowerShell][create_account_with_powershell], or the [Azure CLI][create_account_with_azure_cli].
+
+### Build from Source
+
+First, download the repository to your local folder:
+
+```batch
+git clone https://github.com/Azure/azure-sdk-for-cpp.git
+```
+
+Create a new folder under the root directory of local cloned repo, switch into this folder and run below commands:
+
+Windows:
+
+```batch
+cmake .. -A x64
+cmake --build . --target azure-storage-queues
+```
+
+or Unix:
+
+```batch
+cmake .. -DCMAKE_BUILD_TYPE=Debug
+cmake --build . --target azure-storage-queues
+```
+
+## Key concepts
+
+Common uses of Queue storage include:
+
+- Creating a backlog of work to process asynchronously
+- Passing messages between different parts of a distributed application
+
+Learn more about options for authentication (including Connection Strings, Shared Key, Shared Key Signatures, Active Directory, and anonymous public access) in our [samples](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/storage/azure-storage-queues/samples).
+
+### Thread safety
+
+We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/cpp_introduction.html#thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
+
+### Additional concepts
+
+Client Options | [Accessing the response](https://github.com/Azure/azure-sdk-for-cpp#response-t-model-types) | [Long-running operations](https://github.com/Azure/azure-sdk-for-cpp#long-running-operations) | Handling failures
+
+## Examples
+
+### Send messages
+
+```C++
+const std::string connectionString = "<connection_string>";
+const std::string queueName = "sample-queue";
+
+auto sharedKeyCredential = std::make_shared<StorageSharedKeyCredential>(accountName, accountKey);
```

Comment:
You defined connection string constant above, can we use `CreateFromConnectionString`?